cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -1025,7 +1107,7 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
                 | Opcode::Istore32Complex => {
                     assert!(
```

Comment:
Can this ever fail for valid input?

Reply:
In theory, yes: the complex variants can take as many inputs as the user wants, and the documentation says it'll add all the inputs together to form the effective address. In practice, `postopt` I think is the only creator of these instruction variants, and will only use up to 3 inputs together, to map to what the x86 architecture can do.